diff -p -up multipath-tools-0.4.8/libmultipath/alias.c.binding_error multipath-tools-0.4.8/libmultipath/alias.c
--- multipath-tools-0.4.8/libmultipath/alias.c.binding_error	2007-08-02 23:05:37.000000000 +0200
+++ multipath-tools-0.4.8/libmultipath/alias.c	2009-07-19 11:58:32.000000000 +0200
@@ -202,7 +202,7 @@ lookup_binding(FILE *f, char *map_wwid, 
 			continue;
 		if (sscanf(alias, "mpath%d", &curr_id) == 1 && curr_id >= id)
 			id = curr_id + 1;
-		wwid = strtok(NULL, " \t");
+		wwid = strtok(NULL, "");
 		if (!wwid){
 			condlog(3,
 				"Ignoring malformed line %u in bindings file",
