--- multipath-tools-0.4.8/kpartx/kpartx.c.ext_part	2007-08-02 23:05:37.000000000 +0200
+++ multipath-tools-0.4.8/kpartx/kpartx.c	2009-08-17 15:54:16.000000000 +0200
@@ -487,7 +487,7 @@
 						continue;
 
 					/* Skip all simple slices */
-					if (k < 0)
+					if (slices[j].container == 0)
 						continue;
 
 					/* Check container slice */
@@ -505,7 +505,7 @@
 					if (safe_sprintf(params, "%d:%d %lu",
 							 slices[k].major,
 							 slices[k].minor,
-							 (unsigned long)slices[j].start)) {
+							 (unsigned long)(slices[j].start - slices[k].start))) {
 						fprintf(stderr, "params too small\n");
 						exit(1);
 					}
