--- multipath-tools-0.4.9-20121222/kpartx/Makefile.0001~	2012-12-22 09:40:59.000000000 +0100
+++ multipath-tools-0.4.9-20121222/kpartx/Makefile	2013-01-05 02:38:39.374455389 +0100
@@ -15,13 +15,20 @@ endif
 LDFLAGS = -ldevmapper
 OBJS = bsd.o dos.o kpartx.o solaris.o unixware.o dasd.o sun.o \
        gpt.o mac.o crc32.o lopart.o xstrncpy.o devmapper.o
+SRCS = $(subst .o,.c,$(OBJS))
 EXEC = kpartx
 
 all: $(EXEC)
 
+ifndef WHOLE_PROGRAM
 $(EXEC): $(OBJS)
 	$(CC) $(OBJS) -o $(EXEC) $(LDFLAGS)
 	$(GZIP) $(EXEC).8 > $(EXEC).8.gz
+else
+$(EXEC):
+	$(CC) $(WHOLEFLAGS) $(SRCS) $(CFLAGS) -o $(EXEC) $(LDFLAGS)
+	$(GZIP) $(EXEC).8 > $(EXEC).8.gz
+endif
 
 install: $(EXEC) $(EXEC).8
 	$(INSTALL_PROGRAM) -d $(DESTDIR)$(bindir)
--- multipath-tools-0.4.9-20121222/Makefile.inc.0001~	2012-12-22 09:40:59.000000000 +0100
+++ multipath-tools-0.4.9-20121222/Makefile.inc	2013-01-05 02:38:57.573714202 +0100
@@ -38,6 +38,8 @@ mpathpersistdir = $(TOPDIR)/libmpathpers
 GZIP        = /bin/gzip -9 -c
 INSTALL_PROGRAM = install
 
+WHOLEFLAGS   = -Wl,--no-warn-common -flto -fwhole-program
+
 ifndef RPM_OPT_FLAGS
 	RPM_OPT_FLAGS = -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4
 endif
--- multipath-tools-0.4.9-20121222/multipathd/Makefile.0001~	2012-12-22 09:40:59.000000000 +0100
+++ multipath-tools-0.4.9-20121222/multipathd/Makefile	2013-01-05 02:38:39.378455446 +0100
@@ -20,6 +20,7 @@ LDFLAGS += -lpthread -ldevmapper -lreadl
 # object files
 #
 OBJS = main.o pidfile.o uxlsnr.o uxclnt.o cli.o cli_handlers.o
+SRCS = $(subst .o,.c,$(OBJS))
 
 
 #
@@ -27,9 +28,15 @@ OBJS = main.o pidfile.o uxlsnr.o uxclnt.
 #
 all : $(EXEC)
 
+ifndef WHOLE_PROGRAM
 $(EXEC): $(OBJS)
 	$(CC) $(CFLAGS) $(OBJS) $(LDFLAGS) -o $(EXEC)
 	$(GZIP) $(EXEC).8 > $(EXEC).8.gz
+else
+$(EXEC):
+	$(CC) $(WHOLEFLAGS) $(SRCS) $(CFLAGS) -o $(EXEC) $(LDFLAGS)
+	$(GZIP) $(EXEC).8 > $(EXEC).8.gz
+endif
 
 install:
 	$(INSTALL_PROGRAM) -d $(DESTDIR)$(bindir)
