--- ./path_priority/pp_netapp/pp_netapp.c.tv	2007-10-22 21:05:57.000000000 +0200
+++ ./path_priority/pp_netapp/pp_netapp.c	2007-10-22 21:15:22.000000000 +0200
@@ -53,7 +53,8 @@ static void process_sg_error(struct sg_i
 	if (io_hdr->sb_len_wr > 0) {
 		fprintf(stderr, "- SCSI sense data: ");
 		for (i=0; i<io_hdr->sb_len_wr; i++) {
-			fprintf(stderr, "0x%02x ", io_hdr->sbp[i]);
+				char* ptr = io_hdr->sbp;
+			fprintf(stderr, "0x%02x ", ptr[i]);
 		}
 		fprintf(stderr, "\n");
 	}
