diff -ru multipath-tools-0.4.8.orig/Makefile.inc multipath-tools-0.4.8/Makefile.inc
--- multipath-tools-0.4.8.orig/Makefile.inc	2007-08-02 23:05:37.000000000 +0200
+++ multipath-tools-0.4.8/Makefile.inc	2009-07-28 08:52:36.000000000 +0200
@@ -34,7 +34,7 @@
 CHECKERSLIB = $(checkersdir)/libcheckers
 MULTIPATHLIB = $(multipathdir)/libmultipath
 
-INSTALL_PROGRAM = install -s
+INSTALL_PROGRAM = install -D -s
 
 OPTFLAGS     = -pipe -g -Wall -Wunused -Wstrict-prototypes
 CFLAGS	     = $(OPTFLAGS)
Only in multipath-tools-0.4.8: Makefile.inc~
diff -ru multipath-tools-0.4.8.orig/path_priority/pp_hp_sw/Makefile multipath-tools-0.4.8/path_priority/pp_hp_sw/Makefile
--- multipath-tools-0.4.8.orig/path_priority/pp_hp_sw/Makefile	2007-08-02 23:05:37.000000000 +0200
+++ multipath-tools-0.4.8/path_priority/pp_hp_sw/Makefile	2009-07-28 08:50:17.000000000 +0200
@@ -14,7 +14,7 @@
 	$(CC) -static -o $(EXEC) $(OBJS)
 
 install: $(EXEC)
-	install -m 755 $(EXEC) $(DESTDIR)$(bindir)/$(EXEC)
+	$(INSTALL_PROGRAM) -m 755 $(EXEC) $(DESTDIR)$(bindir)/$(EXEC)
 
 uninstall:
 	rm $(DESTDIR)$(bindir)/$(EXEC)
Only in multipath-tools-0.4.8/path_priority/pp_hp_sw: Makefile~
